### PR TITLE
Category Share on Thanks page

### PR DESCRIPTION
**Pull request by christian14b**
Being able to share the category page and show different messages depending on the loan sector.

This experiment only work for women, agriculture and education categories. If all loans belong to a different category, default Thanks Page will be shown.

image

### CODE REVIEW

File: src/pages/Lend/LoanChannelCategoryPage.vue [hunk 1]
```diff
@@ -67,10 +67,29 @@ const testCategories = [
...
+		let image = '';
+		let title = '';
+		if (this.$route.query.category_share_version
+			&& ['women', 'education', 'agriculture'].includes(this.$route.params.category)) {
+			image = imageRequire(`./${this.$route.params.category}_share_card.png`);
+			title = `Can you help, ${this.$route.query.lender} `;
+			if (this.$route.params.category === 'women') {
+				title += 'support women around the world?';
+			}
+			if (this.$route.params.category === 'education') {
```
Comment (dyersituations) [inline src/pages/Lend/LoanChannelCategoryPage.vue:85]: These would be more appropriate as `else if`

File: src/pages/Thanks/ThanksPage.vue [hunk 2]
```diff
@@ -208,6 +210,26 @@ export default {
 	},
 	computed: {
 		selectedLoan() {
+			if (this.categoryShareVersion === 'a' || this.categoryShareVersion === 'b') {
+				const loans = [...this.loans];
+				loans.sort((a, b) => {
```
Comment (dyersituations) [inline src/pages/Thanks/ThanksPage.vue:215]: Doesn't need to change for this PR, but this sort code is a great example of a chunk of code with very specific expectations that could be extracted to a util and unit tested.

File: src/components/Thanks/ThanksPageShare.vue [hunk 3]
```diff
@@ -70,11 +77,26 @@
...
-							Can you share this loan with one more person?
+							<template v-if="categoryShareVersion === 'a' && categoryName">
+								<!-- eslint-disable-next-line max-len -->
+								<span class="fs-mask">{{ this.lender.firstName }}</span>, share now to find allies in the fight against economic inequity for {{ this.categoryName }}.
+							</template>
+							<template v-else-if="categoryShareVersion === 'b' && categoryName">
+								More loans like yours mean more opportunities for {{ this.categoryName }}.
+							</template>
+							<template v-else>
+								<!-- eslint-disable-next-line max-len -->
```
Comment (dyersituations) [inline src/components/Thanks/ThanksPageShare.vue:88]: I'm guessing this eslint disable isn't needed.

File: src/components/Thanks/ThanksPageShare.vue [hunk 4]
```diff
@@ -198,6 +220,10 @@ export default {
 			type: String,
 			default: 'a'
 		},
+		categoryShareVersion: {
+			type: String,
+			default: 'c'
```
Comment (dyersituations) [inline src/components/Thanks/ThanksPageShare.vue:225]: Is this default no experiment?

File: src/components/Thanks/ThanksPageShare.vue [hunk 5]
```diff
@@ -277,6 +322,45 @@ export default {
...
+				if (this.categoryName === 'women') {
+					pageBody += ' Your loan will help women access to the funds they need to improve their lives.';
+				}
+				if (['education', 'women'].includes(this.categoryName)) {
+					// eslint-disable-next-line max-len
+					pageBody += `Your loan will help borrowers access the funds they need to invest in ${this.categoryName}.`;
+				}
+				pageBody += ' The more people join our cause, the bigger impact we\'ll make.';
+			}
+			if (this.categoryShareVersion === 'b') {
```
Comment (dyersituations) [inline src/components/Thanks/ThanksPageShare.vue:347]: `else if` makes sense here

File: src/components/Thanks/ThanksPageShare.vue (hunk 4 above)
Comment (christian14b) [reply src/components/Thanks/ThanksPageShare.vue:225]: Yes, we're handling two variants c is default!

File: src/pages/Thanks/ThanksPage.vue [hunk 6]
```diff
@@ -377,6 +399,21 @@ export default {
...
 			}
+
+			// MARS-310 Category Share on Thanks page
+			const categoryShareResult = this.apollo.readFragment({
+				id: 'Experiment:share_ask_copy',
+				fragment: experimentVersionFragment,
+			}) || {};
+
+			this.categoryShareVersion = categoryShareResult?.version;
+			if (this.categoryShareVersion) {
```
Comment (emuvente) [inline src/pages/Thanks/ThanksPage.vue:410]: I think we also need to check if the user has a loan that qualifies for this experiment before we track them.

**Review by emuvente — COMMENTED** [review]
A couple additions needed, but overall it's looking great!

**Comment by christian14b** [comment]
Hey! @mcstover @dyersituations anything left I should fix to merge this?

**Review by mcstover — APPROVED** [review]

File: src/components/Thanks/ThanksPageShare.vue [hunk 7]
```diff
@@ -70,11 +77,25 @@
...
+							<template v-else>
+								Can you share this loan with one more person?
+							</template>
 						</h1>
 						<p class="tw-text-h3 tw-m-0 thanks__base-text">
-							<!-- eslint-disable-next-line max-len -->
-							<span class="fs-mask">{{ this.loan.name }}</span> only needs {{ calculatePeopleQtyToGoal() }} more people to lend $25 and their loan could be fully funded in a matter of hours!
+							<template v-if="categoryShareVersion === 'c' || !categoryName">
+								<!-- eslint-disable-next-line max-len -->
+								<span class="fs-mask">{{ this.loan.name }}</span> only needs {{ calculatePeopleQtyToGoal() }} more people to lend $25 and their loan could be fully funded in a matter of hours!
```
Comment (dyersituations) [inline src/components/Thanks/ThanksPageShare.vue:94]: Do we need to mask the loan name here?

File: src/pages/Lend/LoanChannelCategoryPage.vue [hunk 8]
```diff
@@ -67,10 +67,29 @@ const testCategories = [
...
+const imageRequire = require.context('@/assets/images/category-share-experiment/', true);
+
 export default {
 	name: 'LoanChannelCategoryPage',
 	metaInfo() {
 		this.getMetaInfo();
+		let image = '';
+		let title = '';
+		if (this.$route.query.category_share_version
+			&& ['women', 'education', 'agriculture'].includes(this.$route.params.category)) {
```
[the remark this answers is not in the document]
Comment (dyersituations) [reply src/pages/Lend/LoanChannelCategoryPage.vue:79]: @christian14b I'm not seeing the canonical URL for this page, are there changes needed here?

File: src/pages/Lend/LoanChannelCategoryControl.vue [hunk 9]
```diff
@@ -351,6 +351,13 @@ export default {
 			if (this.$route.query.page && Number(this.$route.query.page) > 1) {
 				url = `${url}?page=${this.$route.query.page}`;
 			}
+			// MARS-310 Category Share on Thanks page
```
Comment (christian14b) [inline src/pages/Lend/LoanChannelCategoryControl.vue:354]: @dyersituations this function overrides the canonical url for `lend-by-category` page, didn't notice after a couple of tests

File: src/pages/Lend/LoanChannelCategoryPage.vue (hunk 8 above)
[the remark this answers is not in the document]
Comment (christian14b) [reply src/pages/Lend/LoanChannelCategoryPage.vue:79]: This change was made here in `handleCanonical` func https://github.com/kiva/ui/pull/4396/files#r1035996058

File: src/components/Thanks/ThanksPageShare.vue (hunk 7 above)
Comment (christian14b) [reply src/components/Thanks/ThanksPageShare.vue:94]: You're right, don't think it's necessary!

**Review by dyersituations — APPROVED** [review]

**Pull request merged**